### PR TITLE
JuliaLintBear.py: Use JuliaRequirement

### LINKED ISSUE #1613: JuliaLintBear: Use JuliaRequirement
`REQUIREMENTS` should be a `JuliaRequirement` instead of a `DistributionRequirement`.

### PR DESCRIPTION
**Pull request by prakashdanish**
Closes #1613

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by supergr35** [comment]
Please check our commit message docs

File: bears/julia/JuliaLintBear.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ class JuliaLintBear:
     on the analysis provided.
     """
     LANGUAGES = {'Julia'}
-    REQUIREMENTS = {DistributionRequirement(apt_get='julia')}
+    REQUIREMENTS = {JuliaRequirement(apt_get='julia')}
```
Comment (naveentata) [inline bears/julia/JuliaLintBear.py]: @prakashdanish No need of `apt_get` in the JuliaRequirement. The arguments should be like `('package name', 'package version')`

**Comment by supergr35** [comment]
Uh-oh! You have have the dreaded merge commit. This is probably because you pushed the rebase(update branch from master) button in github. This is not the proper way of doing this and will break many things. Please do an interactive rebase on the commit before the merge commit and remove the merge commit line(see http://stackoverflow.com/questions/17577409/git-remove-merge-commit-from-history). See https://api.coala.io/en/latest/Developers/Git_Basics.html#rebasing for the proper way to rebase.
(it's okay, we all make mistakes in git ;) )

**Comment by prakashdanish** [comment]
@supergr35 sorry for the mess friend. Followed the instruction and tried a rebase with the commit right before the merge commit and pushed to my local fork repo not sure whether it's okay or not.

**Comment by prakashdanish** [comment]
@naveentata Looking at [this](https://gitlab.com/coala/package_manager/blob/master/dependency_management/requirements/JuliaRequirement.py) I think the version arg shouldn't be provided here, right?

**Comment by prakashdanish** [comment]
@supergr35 is it okay now?

**Comment by supergr35** [comment]
Not quite, you still have a merge commit. You'll wanna do an _interactive rebase_ which you can find in our docs 👍 and also in the links put up above.

**Comment by prakashdanish** [comment]
@supergr35  did the reset and removed the said commit, is it okay now?

**Comment by jayvdb** [comment]
The problem with the builds is fixed.
Please change your commit short message to : JuliaLintBear.py: Use JuliaRequirement
Rebase, and cross your fingers. ;-)

**Comment by jayvdb** [comment]
@prakashdanish , you need to [rebase](https://coala.io/rebase).  That is why it is failing.

**Comment by jayvdb** [comment]
reack 9a6e405

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**